### PR TITLE
Just something to look at: Alternative take on Postgres

### PR DESCRIPTION
**Pull request by krancour**
@arschles asked me to PR this just to get it on your radar.

This isn't mergeable because it was built in parallel with your own postgres module (i.e. not built on top of it... much of this was written before I realized you guys were already working on one). _If_ there's interest in moving forward with this, I'm happy to rebase on master.

The implementation is a bit different from what's in master now: it uses ES6 and the Node SDK for "Azure Database for PostgreSQL". That SDK didn't exist prior to this effort. I generated it (and an SDK for MySQL also) from swagger specs and worked with the [azure-sdk-for-node](https://github.com/Azure/azure-sdk-for-node) team to get those new SDKs merged. Note that they are not yet published to NPM. (I'm working on that still.) These SDKs are promise-based, so you will find much of the code in this implementation is promise-based as well.

More significant than the differences in implementation, however, are the differences in scope. What's in master currently only provisions PostgreSQL servers. I opened issue #114 about this because applications that may need to provision and bind to a database generally expect the database to exist and don't posses the intelligence to begin with a fresh server and create and manage the database themselves. So, in terms of scope, this alternative implementation includes provisioning a database as as well (1:1 server:database). Because applications that bind to the service also shouldn't necessarily need administrative privileges, binding is also different in this implementation. Here, binding provides a new, unique set of credentials that can be revoked upon unbind.

It's also worth highlighting what is _common_ to this implementation and what is in master-- I have re-used your `service.json` and also ensured that my implementation honors the _exact_ same parameters as the implementation that's in master. (You need to provide only one additional provisioning parameter-- `postgressqldbName`.)

fwiw, I assume there's some pride in ownership of the postgres module that's in master now and this is a very different implementation. If we could consider merging this eventually, I'd be happy, but I'd be equally happy if we can build on what's in master now to close all the (perceived) gaps this PR addresses...

Bottom line is I'd love this to get merged, but what I care about most is that we use this PR to stimulate further discussion.

### CODE REVIEW

**Comment by krancour** [comment]
Closing this as no longer relevant.
